LiteDB/Client/Mapper/BsonMapper.cs

```diff
@@ -235,13 +242,30 @@ internal EntityMapper GetEntityMapper(Type type)
         {
             //TODO: needs check if Type if BsonDocument? Returns empty EntityMapper?
 
-            if (!_entities.TryGetValue(type, out EntityMapper mapper))
+            if (_entities.TryGetValue(type, out EntityMapper mapper))
+            {
+                return mapper;
+            }
+
+            lock (_entities)
```

Comment:
You mentioned that is not needed to use `ConcurrentDictionary`, but we need a lock here in order to access and update its values (for both), are you sure isn't better to use that collection and avoid the `lock` here?

Reply:
In the best case (which happens 99% of the time), we don`t lock. Only initially, when first creating the mapping, we need the lock. 

In addidtion to that, this behaviour is the same as the original code before the [bugfix that caused this bug](https://github.com/mbdavid/LiteDB/pull/2493)